src/types.ts

```diff
@@ -0,0 +1,148 @@
+/**
```

Comment:
Overall, I think this file has too many types in it.  We already have the modules for the different renderers so I think we should move the specific types into those which import shared types from here (or `core`?).

Personally, I prefer keeping the types with code, but I haven't worked on a TS project with as much crossover in types between modules before so this approach might be better?

As mentioned above, I've got some ideas to reduce the amount of Renderer specific stuff required so I'll see after I propose those changes if there is any more cleanup here.

Reply:
My personal preference is this way, you can see & compose types really simply. But I'm open to doing it the alternative way.
>As mentioned above, I've got some ideas to reduce the amount of Renderer specific stuff required so I'll see after I propose those changes if there is any more cleanup here.

I agree, lets do it after the rest of the changes.